### PR TITLE
feat: LT-2032 - exhibition details

### PR DESCRIPTION
**Pull request by Mark-Spiers-kainos**

### CODE REVIEW

File: applications/serializers/exhibition_clearance.py [hunk 1]
```diff
@@ -44,14 +51,64 @@ class Meta:
...
+    title = serializers.CharField(
+        max_length=255, allow_null=False, error_messages={"blank": "Enter the exhibition name"},
+    )
+    first_exhibition_date = serializers.DateField(
+        allow_null=False, error_messages={"invalid": "Exhibition start date must include a day, a month and a year"},
+    )
+    required_by_date = serializers.DateField(
         allow_null=False,
-        error_messages={"blank": strings.Applications.MISSING_REFERENCE_NAME_ERROR},
+        error_messages={"invalid": "Date the clearance is needed must include a day, a month and a year"},
```
Comment (BenAndrew-Kainos) [inline applications/serializers/exhibition_clearance.py]: Please move to strings

File: applications/serializers/exhibition_clearance.py [hunk 2]
```diff
@@ -44,14 +51,64 @@ class Meta:
...
+            "title",
+            "first_exhibition_date",
+            "required_by_date",
+            "reason_for_clearance",
+        )
+
+
+class ExhibitionClearanceDetailSerializer(serializers.ModelSerializer):
+    title = serializers.CharField(
+        max_length=255, allow_null=False, error_messages={"blank": "Enter the exhibition name"},
```
Comment (BenAndrew-Kainos) [inline applications/serializers/exhibition_clearance.py]: Please move to strings

File: applications/serializers/exhibition_clearance.py (hunk 1 above)
Comment (BenAndrew-Kainos) [inline applications/serializers/exhibition_clearance.py]: Please move to strings

Comment (rorysproule1) [reply applications/serializers/exhibition_clearance.py]: Fixed

File: applications/serializers/exhibition_clearance.py (hunk 2 above)
Comment (rorysproule1) [reply applications/serializers/exhibition_clearance.py]: Fixed

File: applications/serializers/exhibition_clearance.py (hunk 1 above)
Comment (rorysproule1) [reply applications/serializers/exhibition_clearance.py]: Fixed

File: goods/views.py [hunk 3]
```diff
@@ -242,6 +242,11 @@ def delete(self, request, pk):
         return JsonResponse(data={"status": "Good Deleted"}, status=status.HTTP_200_OK)
 
 
+class GoodItemTypes(APIView):
+    def get(self, request):
+        return JsonResponse(data={"item_types": {choice[0]: choice[1] for choice in ItemType.choices}})
```
Comment (BenAndrew-Kainos) [inline goods/views.py]: This should be in static

File: applications/tests/tests_mod_clearance_submit.py [hunk 4]
```diff
@@ -66,71 +66,30 @@ def test_submit_exhibition_clearance_success(self):
         self.assertIsNotNone(application.consignee)
         self.assertIsNotNone(application.goods.get())
 
-    def test_submit_exhibition_clearance_without_end_user_failure(self):
```
[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply applications/tests/tests_mod_clearance_submit.py]: This is still important!

[the remark this answers is not in the document]
Comment (Mark-Spiers-kainos) [reply applications/tests/tests_mod_clearance_submit.py]: Have introduced tests at end of edit, which covers being not being able to add parties.

File: applications/views/applications.py [hunk 5]
```diff
@@ -595,3 +597,71 @@ def duplicate_f680_clearance_types(self):
...
+        if serializer.is_valid():
+            old_title = application.title
+            old_first_exhibition_date = application.first_exhibition_date
+            old_required_by_date = application.required_by_date
+            old_reason_for_clearance = application.reason_for_clearance
+            case = application.get_case()
+            serializer.save()
+            validated_data = serializer.validated_data
+
+            if validated_data["title"] != old_title:
```
Comment (stephanieboyd) [inline applications/views/applications.py]: Better for this to be separated into functions for each audit entry

**Pull request merged**